zkevm-circuits/src/evm_circuit/util/math_gadget.rs

```diff
@@ -200,6 +201,268 @@ impl<F: FieldExt, const N: usize> AddWordsGadget<F, N> {
     }
 }
 
+/// Construction of 2 256-bit words mutiplication and result (modulo 2**256),
+/// which is useful for opcode MUL, DIV, SDIV and xxxMOD
+#[derive(Clone, Debug)]
+pub(crate) struct MulWordsGadget<F> {
+    multiplier: util::Word<F>,
+    multiplicand: util::Word<F>,
+    product: util::Word<F>,
+    //here we execute a multi-limbs multiplication, see spec or
+    //https://hackmd.io/HL0QhGUeQoSgIBt2el6fHA
+    t0: Cell<F>,
+    t1: Cell<F>,
+    t2: Cell<F>,
+    t3: Cell<F>,
+    v0: [Cell<F>; 9],
+    v1: [Cell<F>; 9],
+}
+
+impl<F: FieldExt> MulWordsGadget<F> {
+    pub(crate) fn construct(
+        cb: &mut ConstraintBuilder<F>,
+        multiplier: util::Word<F>,
+        multiplicand: util::Word<F>,
+    ) -> Self {
+        let product = cb.query_word();
+        let t0 = cb.query_cell();
+        let t1 = cb.query_cell();
+        let t2 = cb.query_cell();
+        let t3 = cb.query_cell();
+        let v0: [Cell<F>; 9] = (0..9)
```

Comment:
you can use `array_init(|_| cb.query_cell());`

Reply:
in fact if all cells in `v0` and `v1` are bytes, you can use `array_init(|_| cb.query_bytel());` instead and then you don't need to add the 256 range check for these cells